Automattic/wp-calypso
# [Emails] Improve email management messaging for non owners

#### Proposed Changes

This change uses the previous changes for the messaging in email comparison, but now in the email management screen.
This email management screen might have 2 different errors.

- When the user is not the owner of the email subscription
- When the user is not the owner of the domain

I've added code to differences them and print the message according to the situation

#### Testing Instructions

Might be a bit difficult to test:

**Scenario 1: Having an email subscription and domain, which belong to your user.**
1. Go to Upgrades -> Emails
2. Chose the right domain (according to the scenario)
3. Check that there is no warning present
4. Check that you are able to add mailboxes

![image](https://user-images.githubusercontent.com/5689927/193028126-6bbfefcc-f13f-46ec-b047-17a03b000123.png)


**Scenario 2: Having an email & domain that does not belong to you.**

You can use Calypso to transfer your domain to another user. This will move the domain and the email subscription
1. Go to Upgrades -> Emails
2. Chose the right domain (according to the scenario)
3. Check that you are presented with a warning
4. Check that you are not able to add mailboxes

![image](https://user-images.githubusercontent.com/5689927/193280162-633253ae-aa84-411f-89c1-334ddc3a937d.png)

**Scenario 3: Having an email (yours) whose domain does not belong to you**

You can use Calypso to transfer your domain to another user. This will move the domain and the email subscription. Also to achieve this situation, please refer to this p2: pbLl1t-LC-p2  and transfer the professional email subscription to you.
1. Go to Upgrades -> Emails
2. Chose the right domain (according to the scenario)
3. Check that you are presented with a warning
4. Check that you are not able to add mailboxes

![image](https://user-images.githubusercontent.com/5689927/193027809-62c3b96b-e283-4120-b1af-ed6462b13d1f.png)

**Scenario 4: Remove all mailboxes form the email account and assert that generic notice with CTA appears.**
![image](https://user-images.githubusercontent.com/5689927/193593669-e52c5f2f-ea99-412c-9c4b-1fea898f312a.png)


#### Pre-merge Checklist






Related to #
